### PR TITLE
Return dummy log when batch job before being added to session manager

### PR DESCRIPTION
**Pull request by zwangsheng**
…sion manager

### _Why are the changes needed?_

As title

Current we open session first then hand this session with batch by SessionManager:
https://github.com/apache/kyuubi/blob/413683507079f8f97217e8099177f0b16bc9192d/kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala#L97-L121

We may face `No local log `exception, when kyuubi server picked this batch job to open session but not handled by SessionManager, due to time gap and code gap.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala [hunk 1]
```diff
@@ -415,6 +415,16 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
         } else if (fe.connectionUrl != metadata.kyuubiInstance) {
           val internalRestClient = getInternalRestClient(metadata.kyuubiInstance)
           internalRestClient.getBatchLocalLog(userName, batchId, from, size)
+        } else if (batchV2Enabled(metadata.requestConf) &&
+          // Aims to cover the case that batch job been picked by current instance
+          // but the batch id not been handled by this SessionManager.
+          // Ref to we open session first then hand the session
+          // and mark operation RUNNING also has code gap with handler session,
+          // we consider RUNNING state too
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala:423]:
```suggestion
          // in batch v2 impl, the operation state is changed from PENDING to RUNNING
          // before being added to SessionManager.
```

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.8

**Comment by pan3793** [comment]
Unfortunately, UT failed after this change.
```
BatchesV2ResourceSuite:
- get local log internal redirection *** FAILED ***
  200 did not equal 404 (BatchesResourceSuite.scala:626)
```

**Comment by zwangsheng** [comment]
Sure @pan3793 open #5307 to fix.

**Pull request closed without merging**